### PR TITLE
Add initial pass at generating wheel

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Add logic to generate a wheel. This is essentially #3, simplified and without the external driver script (since the generated wheel isn't yet usable).

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/pip-drake/4)

### CODE REVIEW

**Review by jwnimmer-tri — APPROVED** [review]
:lgtm:

> Reviewed 1 of 3 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/pip-drake/4#-:-MgMnv_Q2EpjAcBVXTdk:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @BetsyMcPhail and @mwoehlke-kitware)

___
*[image/setup.py, line 43 at r2](https://reviewable.io/reviews/robotlocomotion/pip-drake/4#-MgMnXjF5LI3d4FQ6Qnh:-MgMnXjF5LI3d4FQ6Qni:b5s7ykc) ([raw file](https://github.com/robotlocomotion/pip-drake/blob/f22cfac4e6338ca59ccd2bb26364237b700e4f4c/image/setup.py#L43)):*
> ```Python
> ]
> 
> # Required python packages that will be pip installed along with pydrake
> ```

nit Please add a TODO to investigate pruning this list.  In particular, I think `pygame` is probably spurious.
___
*[image/setup.py, line 93 at r2](https://reviewable.io/reviews/robotlocomotion/pip-drake/4#-MgMnlaP-BqyDCbCQXxC:-MgMnlaP-BqyDCbCQXxD:b-ofux51) ([raw file](https://github.com/robotlocomotion/pip-drake/blob/f22cfac4e6338ca59ccd2bb26364237b700e4f4c/image/setup.py#L93)):*
> ```Python
>         'Topic :: Software Development :: Libraries :: Python Modules'],
>       distclass=BinaryDistribution,
>       license='BSD 3-Clause License',
> ```

nit Please add a TODO to refine our license statement.  Since we are shipping bundled dependencies, we might have to report more information here?

File: image/build-wheel.sh [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
-git clone https://github.com/robotlocomotion/drake
+cp -r -t /wheel/pydrake \
+    /opt/drake/share/doc \
+    /opt/drake/share/drake/.drake-find_resource-sentinel \
+    /opt/drake/share/drake/examples \
+    /opt/drake/share/drake/manipulation \
+    /opt/drake/lib/python*/site-packages/pydrake/*
 
-cd /drake
+cp -r -t /wheel/pydrake/lib \
```
Comment (BetsyMcPhail) [inline image/build-wheel.sh:14]: Is /opt/drake-dependencies/lib/lib*.so* needed too?

**Review by BetsyMcPhail — COMMENTED** [review]
Splitting up the build-wheel and build-drake is a good improvement.

File: image/build-wheel.sh (hunk 1 above)
Comment (mwoehlke-kitware) [reply image/build-wheel.sh:14]: `/opt/drake-dependencies/lib/lib*.so` doesn't *exist* (there are, by design, only `.a`s), so... 🙂

**Comment by mwoehlke-kitware** [comment]
> Please add a TODO to investigate pruning this list. In particular, I think pygame is probably spurious.

Done.

> Please add a TODO to refine our license statement. Since we are shipping bundled dependencies, we might have to report more information here?

Done.

BTW, Reviewable's integration is rather inferior to Github's built-in review. Obviously, when we started using it, it was useful, but now that Github's own review tools have improved significantly, what is the advantage to Reviewable?

File: image/build-wheel.sh (hunk 1 above)
Comment (BetsyMcPhail) [reply image/build-wheel.sh:14]: There's a few so's in there:

libjpeg.so  liblzma.so  libpng16.so      libz.so
liblz4.so   libpng.so   libturbojpeg.so

I think Jeremy found he needed to add `/opt/drake-dependencies/lib` to LD_LIBRARY_PATH ([slack](https://drakedevelopers.slack.com/archives/C028F9T6MSM/p1627963270004100?thread_ts=1627959170.002400&cid=C028F9T6MSM)) ?

Comment (mwoehlke-kitware) [reply image/build-wheel.sh:14]: There shouldn't be. There aren't on my build:
```
# ls /opt/drake-dependencies/lib/*.so*
ls: cannot access '/opt/drake-dependencies/lib/*.so*': No such file or directory
```

Are you at or after 0643d35cedcabda6efedc5a2ae5a38351b43cfda?

Comment (BetsyMcPhail) [reply image/build-wheel.sh:14]: Ah, I was at an older commit.

**Review by BetsyMcPhail — COMMENTED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/pip-drake/4#-:-MgRB21G3fLVhQbacR_P:bg7igkv)* status: 3 of 5 files reviewed, 3 unresolved discussions (waiting on @jwnimmer-tri and @mwoehlke-kitware)

**Comment by mwoehlke-kitware** [comment]
(FYI, only change in the latest force-push is fixing permissions on `build-wheel.sh`.)

**Comment by jwnimmer-tri** [comment]
FYI I haven't had a chance to check back here again, but I'm still fine if you want to merge this.  (My approval upthread still stands.)

**Pull request merged**